### PR TITLE
PIM-5800: Update product import doc

### PR DESCRIPTION
**Pull request by solivier**

### CODE REVIEW

File: reference/import_export/product-export.rst [hunk 1]
```diff
@@ -16,20 +16,41 @@ The product export is defined in ``src/Pim/Bundle/ConnectorBundle/Resources/conf
```
Comment (jjanvier) [inline reference/import_export/product-export.rst]: Please note that the *export* jobs have been widely re-worked in 1.6.

Comment (jjanvier) [inline reference/import_export/product-export.rst]: and maybe explain that now you can export natively in CSV and XLSX

and that you can finely select which products to export

Comment (rybus) [inline reference/import_export/product-export.rst]: Understanding the product export

File: reference/import_export/product-export.rst [hunk 2]
```diff
@@ -1,7 +1,7 @@
-Understand the CSV Product Export
-=================================
+Understand the Product Export
+=============================
 
-We'll discuss here how the CSV product export works.
+We'll discuss here how the product export works.
```
Comment (rybus) [inline reference/import_export/product-export.rst]: This sentence paraphrases the title and thus is useless.

File: reference/import_export/product-import.rst [hunk 3]
```diff
@@ -1,68 +1,81 @@
-Understand the CSV Product Import
-=================================
+Understand the Product Import
+=============================
 
-We'll discuss here how the csv product import works.
+We'll discuss here how the product imports works.
```
Comment (MarieMinasyan) [inline reference/import_export/product-import.rst]: import works or imports work

File: reference/import_export/product-export.rst [hunk 4]
```diff
@@ -40,16 +61,31 @@ All steps service definitions are defined in ``src/Pim/Bundle/ConnectorBundle/Re
...
+            arguments:
+                - 'export'
+                - '@event_dispatcher'
+                - '@akeneo_batch.job_repository'
+                - '@pim_connector.reader.database.product'
+                - '@pim_connector.processor.normalization.product'
+                - '@pim_connector.writer.file.xlsx_product'
+                - 10
 
 An ``ItemStep`` always contains 3 elements, a ``Akeneo\Bundle\BatchBundle\Item\ItemReaderInterface``, a ``Akeneo\Bundle\BatchBundle\Item\ItemProcessorInterface`` and a ``Akeneo\Bundle\BatchBundle\Item\ItemWriterInterface``.
```
Comment (rybus) [inline reference/import_export/product-export.rst]: A bullet-style list would make this clearer.

File: reference/import_export/product-import.rst [hunk 5]
```diff
@@ -138,20 +151,35 @@ The ``Akeneo\Component\Batch\Model\StepExecution`` allows to add information, me
 Product Import Step
 -------------------
 
-The purpose of this step is to read input CSV file, to transform lines to product objects, to validate and save them in the PIM.
+The purpose of this step is to read input file, to transform lines to product objects, to validate and save them in the PIM.
```
Comment (MarieMinasyan) [inline reference/import_export/product-import.rst]: to read an input file

File: reference/import_export/product-import.rst [hunk 6]
```diff
@@ -1,68 +1,81 @@
-Understand the CSV Product Import
-=================================
+Understand the Product Import
```
Comment (rybus) [inline reference/import_export/product-import.rst]: Understanding the product import

File: reference/import_export/product-import.rst (hunk 3 above)
Comment (rybus) [inline reference/import_export/product-import.rst]: Same notice about paraphrasing the title ;)

File: reference/import_export/product-import.rst [hunk 7]
```diff
@@ -1,68 +1,81 @@
...
 
-We'll discuss here how the csv product import works.
+We'll discuss here how the product imports works.
 
 It's a good start to understand the overall architecture and how to re-use or replace some parts.
 
 .. note::
 
-  The import part has been widely re-worked in 1.4. Since 1.6, the old import system is deleted; the following documentation is related to this new system.
+  The import part has been widely re-worked in 1.6. Since 1.6, the old import system is deleted; the following documentation is related to this new system.
```
Comment (jjanvier) [inline reference/import_export/product-import.rst]: explain that we can now import natively both in CSV and in XLSX

Comment (rybus) [inline reference/import_export/product-import.rst]: The import [...] in 1.6. Old import system has been removed, please refer to previous versions of this page if needed.

File: reference/import_export/product-import.rst [hunk 8]
```diff
@@ -1,68 +1,81 @@
...
+                - '@akeneo_batch.job_repository'
+                -
+                    - '@pim_connector.step.charset_validator'
+                    - '@pim_connector.step.xlsx_product.import'
+                    - '@pim_connector.step.xlsx_product.import_associations'
+            tags:
+                - { name: akeneo_batch.job, connector: '%pim_connector.connector_name.xlsx%', type: '%pim_connector.job.import_type%' }
 
-We can also count 3 defined steps, ``validation``, ``import`` and ``import_associations``.
+With the ``type`` parameter, we can see that this job is an import.
```
Comment (rybus) [inline reference/import_export/product-import.rst]: Maybe you can list here other options for `type` just for the record.

File: reference/import_export/product-import.rst [hunk 9]
```diff
@@ -1,68 +1,81 @@
...
 
 Charset Validation Step
 -----------------------
 
 The purpose of this step is to validate that the input file has the expected encoding (UTF-8 by default).
```
Comment (rybus) [inline reference/import_export/product-import.rst]: the input file is properly encoded (default: UTF-8)

File: reference/import_export/product-import.rst (hunk 5 above)
Comment (rybus) [inline reference/import_export/product-import.rst]: to read the input file/input files [...]

File: reference/import_export/product-import.rst [hunk 10]
```diff
@@ -160,87 +188,61 @@ We provide here specific implementations for these elements, the services are de
...
-        ProductBuilderInterface $builder,
-        ObjectUpdaterInterface $updater,
-        ValidatorInterface $validator,
-        ObjectDetacherInterface $detacher,
-        ProductFilterInterface $productFilter,
-        AttributeLocalizedConverterInterface $localizedConverter
-    ) {
-        // ...
-    }
+    This step is able to extract a Zip archive which contains a file and a folder for related images or files. The file has to use relative paths to reference the files.
```
Comment (rybus) [inline reference/import_export/product-import.rst]: zip / ZIP

Comment (rybus) [inline reference/import_export/product-import.rst]: which contain a CSV file for products and next to it a folder containing images. The product file refers to images using relatives paths.

File: reference/import_export/product-import.rst [hunk 11]
```diff
@@ -303,6 +305,53 @@ The service uses the class ``Akeneo\Component\Localization\Localize\AttributeCon
 
     Read the cookbook to add your own localizer  :doc:`/cookbook/localization/index`
 
+Product Processor
+-----------------
+
+This element receives items one by one, fetches or creates the related product, updates and validates it.
```
Comment (rybus) [inline reference/import_export/product-import.rst]: recieves (US)

Comment (rybus) [inline reference/import_export/product-import.rst]: creates (or fetches if it already exists) the related product, updates it and validates it.

File: reference/import_export/product-import.rst [hunk 12]
```diff
@@ -303,6 +305,53 @@ The service uses the class ``Akeneo\Component\Localization\Localize\AttributeCon
...
+The class ``Pim\Component\Connector\Processor\Denormalization\ProductProcessor`` mainly delegates the operations to different technical and business services.
+
+.. code-block:: php
+
+    /**
+     * @param IdentifiableObjectRepositoryInterface $repository    product repository
+     * @param ProductBuilderInterface               $builder       product builder
+     * @param ObjectUpdaterInterface                $updater       product updater
+     * @param ValidatorInterface                    $validator     product validator
+     * @param ObjectDetacherInterface               $detacher      detacher to remove it from UOW when skip
```
Comment (rybus) [inline reference/import_export/product-import.rst]: when skipping an item.

File: reference/import_export/product-export.rst (hunk 1 above)
Comment (solivier) [inline reference/import_export/product-export.rst]: @jjanvier it's done in the Reader part

File: reference/import_export/product-import.rst (hunk 8 above)
Comment (solivier) [inline reference/import_export/product-import.rst]: there is another example for export in the export page

File: reference/import_export/product-import.rst [hunk 13]
```diff
@@ -138,20 +150,35 @@ The ``Akeneo\Component\Batch\Model\StepExecution`` allows to add information, me
 Product Import Step
 -------------------
 
-The purpose of this step is to read input CSV file, to transform lines to product objects, to validate and save them in the PIM.
+The purpose of this step is to read an input file, to transform lines to product objects, to validate and save them in the PIM.
```
Comment (rybus) [inline reference/import_export/product-import.rst]: into product objects

File: reference/import_export/product-export.rst [hunk 14]
```diff
@@ -1,13 +1,12 @@
...
+Understanding the Product Export
+================================
 
 It's a good start to understand the overall architecture and how to re-use or replace some parts.
+You can now natively export data into CSV and XLSX format.
 
 .. note::
 
-  Please note that the imports jobs have been widely re-worked in 1.6.
+  Please note that the imports jobs have been widely re-worked in 1.6. Old import system has been removed, please refer to previous versions of this page if needed.
```
Comment (MarieMinasyan) [inline reference/import_export/product-export.rst]: The old

File: reference/import_export/product-import.rst [hunk 15]
```diff
@@ -160,87 +187,61 @@ We provide here specific implementations for these elements, the services are de
...
-        ProductBuilderInterface $builder,
-        ObjectUpdaterInterface $updater,
-        ValidatorInterface $validator,
-        ObjectDetacherInterface $detacher,
-        ProductFilterInterface $productFilter,
-        AttributeLocalizedConverterInterface $localizedConverter
-    ) {
-        // ...
-    }
+    This step is able to extract a zip archive which contain a file for products and next to it a folder containing images. The product file refers to images using relatives paths.
```
Comment (rybus) [inline reference/import_export/product-import.rst]: which conain**s**

File: reference/import_export/product-import.rst [hunk 16]
```diff
@@ -303,6 +304,53 @@ The service uses the class ``Akeneo\Component\Localization\Localize\AttributeCon
 
     Read the cookbook to add your own localizer  :doc:`/cookbook/localization/index`
 
+Product Processor
+-----------------
+
+This element receives items one by one, creates (or fetches if it already exists) the related product, updates it and validates it.
```
Comment (rybus) [inline reference/import_export/product-import.rst]: recieves

File: reference/import_export/product-export.rst [hunk 17]
```diff
@@ -40,18 +60,37 @@ All steps service definitions are defined in ``src/Pim/Bundle/ConnectorBundle/Re
...
+                - 'export'
+                - '@event_dispatcher'
+                - '@akeneo_batch.job_repository'
+                - '@pim_connector.reader.database.product'
+                - '@pim_connector.processor.normalization.product'
+                - '@pim_connector.writer.file.xlsx_product'
+                - 10
 
-An ``ItemStep`` always contains 3 elements, a ``Akeneo\Bundle\BatchBundle\Item\ItemReaderInterface``, a ``Akeneo\Bundle\BatchBundle\Item\ItemProcessorInterface`` and a ``Akeneo\Bundle\BatchBundle\Item\ItemWriterInterface``.
+An ``ItemStep`` always contains 3 elements:
```
Comment (MarieMinasyan) [inline reference/import_export/product-export.rst]: This isn't completely true. The charset validation step has only 1 element.

File: reference/import_export/product-import.rst [hunk 18]
```diff
@@ -1,68 +1,80 @@
...
+Understanding the Product Import
+================================
 
 It's a good start to understand the overall architecture and how to re-use or replace some parts.
+You can now natively import data into CSV and XLSX format.
 
 .. note::
 
-  The import part has been widely re-worked in 1.4. Since 1.6, the old import system is deleted; the following documentation is related to this new system.
+  The import part has been widely re-worked in 1.6. Since 1.6. Old import system has been removed, please refer to previous versions of this page if needed.
```
Comment (MarieMinasyan) [inline reference/import_export/product-import.rst]: The old

File: src/Acme/Bundle/CsvCleanerConnectorBundle/ArrayConverter/StandardToFlat/Product.php [hunk 19]
```diff
@@ -0,0 +1,29 @@
...
+    public function __construct(ArrayConverterInterface $productConverter)
+    {
+        $this->productConverter = $productConverter;
+    }
+
+    public function convert(array $item, array $options = [])
+    {
+        $convertedItem = $this->productConverter->convert($item, $options);
+
+        // clean sku
```
Comment (rybus) [inline src/Acme/Bundle/CsvCleanerConnectorBundle/ArrayConverter/StandardToFlat/Product.php]: cleaned sku // cleans the sku

**Comment by rybus** [comment]
GTM once comments taken into account. You did a great job!

File: reference/import_export/product-import.rst (hunk 15 above)
Comment (MarieMinasyan) [inline reference/import_export/product-import.rst]: contains

**Pull request merged**